packages/react/src/hooks/use-last-factor.ts

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useMemo, useCallback } from "react";
+import { isBrowser } from "@slashid/react-primitives";
+import { Handle, isFactor } from "../domain/types";
+import { useConfiguration } from "./use-configuration";
+import { useSlashID } from "./use-slash-id";
+import { Factor } from "@slashid/slashid";
+
+export const STORAGE_LAST_FACTOR_KEY = (oid: string) =>
+  `@slashid/LAST_FACTOR/${oid}`;
+
+type UseLastFactor = () => {
+  lastFactor: Factor | undefined;
+};
+
+type SuccessEvent = {
+  authenticationFactor: Handle | undefined;
+};
+
+export const useLastFactor: UseLastFactor = () => {
+  const { storeLastFactor } = useConfiguration();
+  const { sid } = useSlashID();
+
+  const lastFactor = useMemo(() => {
+    if (!isBrowser()) {
+      return undefined;
+    }
+
+    try {
+      const storedFactor = window.localStorage.getItem(
+        STORAGE_LAST_FACTOR_KEY(sid?.oid ?? "")
+      );
+      if (!storeLastFactor || !storedFactor) {
+        return undefined;
+      }
+
+      return JSON.parse(storedFactor);
+    } catch {
+      return undefined;
+    }
+  }, [storeLastFactor, sid]);
+
+  const handler = useCallback(
+    ({ authenticationFactor }: SuccessEvent) => {
+      if (!isBrowser()) {
+        return;
+      }
+
+      if (!isFactor(authenticationFactor)) {
```

Comment:
Why do we need to make this check? I would assume that the `SuccessEvent` would always have this value 🤔

Reply:
I mean it's just a bit of defensive programming, very unlikely that there's some garbage in there but does it hurt to have it here?